### PR TITLE
chore: Implement Release Please GitHub actions

### PR DESCRIPTION
**Pull request by alimpens**

### CODE REVIEW

File: documentation/publishing.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # Publishing
 
-We use Lerna to create changelogs and releases for all our packages and to publish them to npm.
+We use a Release Please GitHub action to create changelog and release PRs for all our packages.
```
Comment (VincentSmedinga) [inline documentation/publishing.md:3]:
```suggestion
We use a [Release Please](https://github.com/googleapis/release-please) GitHub action to create a changelog and release PRs for all our packages.
```

File: documentation/publishing.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+When the release PR is merged, that same action publishes the new release to npm, GitHub releases and deploys it to our main Storybook instance.
```
Comment (VincentSmedinga) [inline documentation/publishing.md:4]:
```suggestion
When the release PR is merged, that same action publishes the new release to npm, creates a release on GitHub, and deploys it to our Storybook environment.
```

File: documentation/publishing.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
 
-Only members of the `@amsterdam` group can publish to npm.
+Only the [maintainers](./documentation/maintainers.md) can release new versions of our packages.
```
Comment (VincentSmedinga) [inline documentation/publishing.md:6]:
```suggestion
The [maintainers](./documentation/maintainers.md) can release new versions of our packages.
```

File: documentation/publishing.md [hunk 4]
```diff
@@ -27,28 +28,25 @@ This will cause a major version bump in both packages on release and add its des
...
-2. Run `npm run publish`
-
-We also want to update our Storybook so it’s in sync with the new content of the packages on npm.
-We do this by merging `develop` into `main`.
-Note: the GitHub GUI doesn’t seem to allow you to perform fast-forward merges, so you should do this from your CLI.
-
-1. `git checkout main`
-2. `git merge --ff-only develop`
-3. `git push`
+1. Locally merge the latest version of `develop` into `main` using a fast-forward merge and push to remote:
```
Comment (VincentSmedinga) [inline documentation/publishing.md:31]:
```suggestion
1. Locally merge the latest version of `develop` into `main` using a fast-forward merge, and push to the remote:
```

File: documentation/publishing.md [hunk 5]
```diff
@@ -27,28 +28,25 @@ This will cause a major version bump in both packages on release and add its des
...
+     git checkout main
+     git pull
+     git merge --ff-only origin/develop
+     git push
+   ```
+
+2. This triggers an action which creates a release PR.
+   Check this PR and merge it into `main`.
+   This triggers the same action, to publish the release to npm and GitHub releases.
+   It also deploys the released version to our main Storybook instance.
```
Comment (VincentSmedinga) [inline documentation/publishing.md:43]:
```suggestion
2. This triggers a GitHub Action, which creates a release PR.
   Review, approve and merge this PR.
   The same Action will then publish the release to npm and GitHub.
   It also deploys the released version to our Storybook environment.
```

File: documentation/publishing.md [hunk 6]
```diff
@@ -27,28 +28,25 @@ This will cause a major version bump in both packages on release and add its des
...
+3. When the release actions have completed, there’s a new release commit on `main`.
+   Locally merge this commit back into `develop` and push to remote:
```
Comment (VincentSmedinga) [inline documentation/publishing.md:45]:
```suggestion
3. When complete, the Action adds a new release commit to `main`.
   Locally merge this commit back into `develop` and push it to the remote:
```

**Review by VincentSmedinga — COMMENTED** [review]
Just some suggestions for the docs and naming. I can’t oversee the implementation very well – looks neat, well documented, tested. Thanks!

File: documentation/publishing.md (hunk 1 above)
Comment (alimpens) [reply documentation/publishing.md:3]: Good idea, but I'll link directly to the action itself

File: documentation/publishing.md (hunk 2 above)
Comment (alimpens) [reply documentation/publishing.md:4]: Hm, we have several Storybook environments though. The reason I wrote 'main' is to clarify that it goes to the main one

**Review by VincentSmedinga — APPROVED** [review]

**Review by VincentSmedinga — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+          workflow: "lint-test.yml"
+          name: storybook-react
+          # React storybook is deployed to the /storybook-react subdirectory.
+          # In the main docs storybook, the contents of /storybook-react
+          # are also shown using Storybook composition.
+          path: dist/storybook/storybook-react
+        if: ${{ steps.release.outputs.releases_created }}
+
+      - name: Deploy to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@v4
```
Comment (dlnr) [inline .github/workflows/publish.yml:98]: Jeffrey had to add `dawidd6/action-download-artifact` to the whitelist, maybe we need to add this one (and perhaps others) too.

**Review by dlnr — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 7 above)
Comment (dlnr) [reply .github/workflows/publish.yml:98]: Oh wait it's in there!

**Pull request merged**